### PR TITLE
Ingela/ssl/export key materials doc/otp 19016

### PR DESCRIPTION
**Pull request by IngelaAndin**
Document export_key_materials/4,5 and deprecate prf/5

### CODE REVIEW

File: lib/stdlib/src/otp_internal.erl [hunk 1]
```diff
@@ -259,6 +261,8 @@ obsolete(pg2, _, _) ->
 obsolete(_,_,_) -> no.
 
 -dialyzer({no_match, obsolete_type/3}).
+obsolete_type(ssl, prf_random, 0) ->
+    {deprecated, "Only used in deprecated function prf/5 will no longer bee needed."};
```
Comment (u3s) [inline lib/stdlib/src/otp_internal.erl:265]: typo: bee -> be

File: lib/stdlib/src/otp_internal.erl [hunk 2]
```diff
@@ -64,6 +64,8 @@ obsolete(net, sleep, 1) ->
     {deprecated, "use 'receive after T -> ok end' instead"};
 obsolete(queue, lait, 1) ->
     {deprecated, "use queue:liat/1 instead"};
+obsolete(ssl, prf, 5) ->
+    {deprecated, "Use export_key_materials/4 instead. Note that in OTP 28 the 'testing' way of calling this function will no longer be supported."};
```
Comment (u3s) [inline lib/stdlib/src/otp_internal.erl:68]: maybe wrap line. you reached 147 characters here.

File: lib/ssl/src/ssl.erl [hunk 3]
```diff
@@ -3165,7 +3171,7 @@ directing it to use a specific value from the session security parameters.
 
 > #### Note {: .info }
 >
-> This function is replaced by `export_key_materials/4`, since OTP-26.3, which
+> This function is replaced by `export_key_materials/4`, official documented API function since OTP-27, which
```
Comment (u3s) [inline lib/ssl/src/ssl.erl:3174]: Rickard: "OTP-27" -> "OTP 27"

File: lib/ssl/src/ssl.erl [hunk 4]
```diff
@@ -121,6 +121,12 @@ information about the supported standards see [ssl(6)](ssl_app.md).
 %% Tracing
 -export([handle_trace/3]).
 
+-deprecated([{prf,5,"Use export_key_materials/4 instead. "
+              "Note that in OTP 28 the 'testing' way of calling this function will no longer be supported."
+              }]).
+
+-deprecated_type([{prf_random, 0,"Only used in deprecated function prf/5 will no longer bee needed."}]).
```
Comment (u3s) [inline lib/ssl/src/ssl.erl:128]: typo: bee -> be (check it globally pls, looks like copy/pasted)

**Review by u3s — APPROVED** [review]

File: lib/stdlib/src/otp_internal.erl (hunk 2 above)
Comment (IngelaAndin) [reply lib/stdlib/src/otp_internal.erl:68]: This is automatically generated by the script.

File: lib/stdlib/src/otp_internal.erl (hunk 1 above)
Comment (IngelaAndin) [reply lib/stdlib/src/otp_internal.erl:265]: fixed

File: lib/ssl/src/ssl.erl (hunk 4 above)
Comment (IngelaAndin) [reply lib/ssl/src/ssl.erl:128]: fixed

File: lib/ssl/src/ssl.erl (hunk 3 above)
Comment (IngelaAndin) [reply lib/ssl/src/ssl.erl:3174]: Fixed all ocurrences

**Comment by IngelaAndin** [comment]
@dgud @rickard-green please re-check

File: lib/ssl/src/ssl.erl [hunk 5]
```diff
@@ -3165,7 +3171,7 @@ directing it to use a specific value from the session security parameters.
 
 > #### Note {: .info }
 >
-> This function is replaced by `export_key_materials/4`, since OTP-26.3, which
+> This function is replaced by `export_key_materials/4`, official documented API function since OTP 27, which
 > is equivalent to
 > [`prf(TLSSocket, master_secret, Label, [client_random, server_random, Context], WantedLength)`](`prf/5`)
```
Comment (garazdawi) [inline lib/ssl/src/ssl.erl:3176]:
```suggestion
> [`prf(TLSSocket, master_secret, Label, [client_random, server_random, Context], WantedLength)`](`prf/5`).
```

Comment (IngelaAndin) [reply lib/ssl/src/ssl.erl:3176]: fixed

**Pull request merged**